MPF.Core/UI/ViewModels/CreateIRDViewModel.cs

```diff
@@ -0,0 +1,1071 @@
+            _logPath = string.Empty;
+
+            _keyPath = string.Empty;
+            _hexKey = string.Empty;
+            _key = null;
+            //_keyStatus = "Will attempt to pull Encryption Key from redump.org";
+            _keyStatus = "Cannot create an IRD without a key";
+
+            _picPath = string.Empty;
+            _layerbreakString = string.Empty;
+            _picString = string.Empty;
+            _pic = null;
+            _picStatus = "Will generate a PIC assuming a Layerbreak of 12219392";
+
+            InputPathTextBoxEnabled = true;
+            InputPathBrowseButtonEnabled = true;
+            LogPathTextBoxEnabled = true;
+            LogPathNotProvided = true;
+            LogPathBrowseButtonEnabled = true;
+            KeyPathTextBoxEnabled = true;
+            KeyPathBrowseButtonEnabled = true;
+            HexKeyTextBoxEnabled = true;
+            PICPathTextBoxEnabled = true;
+            PICPathBrowseButtonEnabled = true;
+            PICTextBoxEnabled = true;
+            LayerbreakTextBoxEnabled = true;
+            CreateIRDButtonEnabled = false;
+
+            EnableEventHandlers();
+        }
+
+        #region Property Updates
+
+        /// <summary>
+        /// Trigger a property changed event
+        /// </summary>
+        private void TriggerPropertyChanged(string propertyName)
+        {
+            // Disable event handlers temporarily
+            bool cachedCanExecuteSelectionChanged = CanExecuteSelectionChanged;
+            DisableEventHandlers();
+
+            // If the property change event is initialized
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+
+            // Reenable event handlers, if necessary
+            if (cachedCanExecuteSelectionChanged) EnableEventHandlers();
+        }
+
+        #endregion
+
+        #region UI Commands
+
+        /// <summary>
+        /// Determine if the Create IRD button should be enabled
+        /// </summary>
+        private bool ShouldEnableCreateIRDButton()
+        {
+            return !string.IsNullOrEmpty(this.InputPath)
+                && File.Exists(this.InputPath);
+        }
+
+        /// <summary>
+        /// Change the currently selected input path
+        /// </summary>
+        public void ChangeInputPath()
+        {
+            this.CreateIRDButtonEnabled = ShouldEnableCreateIRDButton();
+        }
+
+        /// <summary>
+        /// Change the currently selected .getkey.log path
+        /// </summary>
+        public void ChangeLogPath()
+        {
+            if (string.IsNullOrEmpty(this.LogPath))
+            {
+                // No .getkey.log file provided: Reset Key and PIC sections
+                LogPathNotProvided = true;
+
+                Key = null;
+                //KeyStatus = "Will attempt to pull Encryption Key from redump.org"; // Use this when redump key pulling is implemented
+                _keyStatus = "Cannot create an IRD without a key";
```

Comment:
Did you mean to assign to `KeyStatus` here instead? It won't trigger a UI update otherwise.

Reply:
Last minute copy paste mistake...thanks